crates/ruff/src/rules/flake8_bugbear/rules/reuse_of_groupby_generator.rs

```diff
@@ -53,6 +64,9 @@ struct GroupNameFinder<'a> {
     /// A flag indicating that the visitor is inside a nested `for` or `while`
     /// loop or inside a `dict`, `list` or `set` comprehension.
     nested: bool,
+    /// A flag indicating the state of the visitor inside a mutually exclusive
+    /// statement such as `if ... elif ... else ...` or a `match` statement.
+    mutually_exclusive_state: MutuallyExclusiveState,
```

Comment:
My intuition (without reading the code deeply) is that this might need to be handled as a stack, that we push and pop from as we enter and leave branchable code?

Reply:
Yes, I'm thinking of using a stack but it's getting a bit complicated so might take some time to resolve this.